ocp_resources/node_network_configuration_policy.py

```diff
@@ -322,10 +323,37 @@ def _absent_interface(self):
         if self.ports:
             self.add_ports()
 
+        initial_transition_time = self._get_nncp_configured_last_transition_time()
         ResourceEditor(
             patches={self: {"spec": {"desiredState": {"interfaces": self.desired_state["interfaces"]}}}}
         ).update()
 
+        # If the setup NNCP failed - we don't need to verify its teardown status time is updated.
+        if initial_transition_time:
+            self._wait_for_nncp_status_update(initial_transition_time=initial_transition_time)
+
+    def _get_nncp_configured_last_transition_time(self):
+        for condition in self.instance.status.conditions:
+            if (
+                condition["type"] == NodeNetworkConfigurationPolicy.Conditions.Type.AVAILABLE
+                and condition["status"] == "True"
+                and condition["reason"] == NodeNetworkConfigurationPolicy.Conditions.Reason.SUCCESSFULLY_CONFIGURED
+            ):
+                return condition["lastTransitionTime"]
+
+    @retry(
+        wait_timeout=TIMEOUT_1MINUTE,
+        sleep=TIMEOUT_5SEC,
+    )
+    def _wait_for_nncp_status_update(self, initial_transition_time):
+        date_format = "%Y-%m-%dT%H:%M:%SZ"
+        formatted_initial_transition_time = datetime.strptime(initial_transition_time, date_format)
+        return any(
```

Comment:
You use here `any` but it will be always only one match right?
if so please operate only on the type you need (similar to your above code for example)

Reply:
@sbahar619 Please note, as you requested this change:
`any` doesn't fit here, because there is only one state that is valid (if `type==AVAILABLE` and `lastTransitionTime > formatted_initial_time`). There is not a set of conditions, were it's enough that one of them is true.